src/app.ts

```diff
@@ -30,8 +39,42 @@ app.all("/signed-image-url", (req: express.Request, res: express.Response) => {
     profile = req.body.profile;
   }
 
-  const signedUrl = format(url, salt, profile);
-  res.send({ signedUrl });
+  const panda = getPanda();
+  panda.verify(req.headers.cookie || "").then((panAuthResult) => {
+    if (panAuthResult.status === 'Authorised') {
+      try {
+        const signedUrl = format(url, salt, profile);
+        res.send({ signedUrl });
+      } catch (ex) {
+        res.status(500).send({ error: "Error signing url", ex: ex });
+      }
+    } else {
+      res.status(403).send({ error: "Not authorised by pan-domain login" }); 
+    }
+  }).catch((ex) => {
+    res.status(500).send({ error: "Pan domain auth error", ex: ex });
+  });
+
+  
+});
+
+app.get("/pandacheck", (req: express.Request, res: express.Response) => {
+  const panda = getPanda();
+  panda.verify(req.headers.cookie || "").then(({ status, user }) => {
+		if (status === 'Authorised') {
+			// TODO MRB: remove once @guardian/pan-domain-node supports an API not including the refresher
+			panda.stop();
+
+			res.send({ authorised: true, user: user });
+		} else {
+			res.send({ authorised: false });
+		}
+	})
+	.catch(ex => {
+    console.error("pan-domain-node error ", ex);
+		panda.stop();
+		res.send({ error: "pan-domain-node error" });
+	});
```

Comment:
I _think_ the indentation is a little off here, which is a good reminder for us to setup prettier/linting on this project. https://github.com/guardian/configs is probably a good place to start! (Separate PR though).